### ISSUE TITLE
[k8s 1.8.2] ipvs table was not updated when adding new service

### ISSUE THREAD

**Issue by gyliu513** [kubernetes/kubernetes#56140]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>

/kind bug

> /kind feature

**What happened**:

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
I was using kubernetes 1.8.2 and calico 2.4 with ipvs.

This is my proxy configuration
```
        "name": "proxy",
        "image": "kubernetes:v1.8.2-ee",
        "imagePullPolicy": "IfNotPresent",
        "command": [
          "/hyperkube",
          "proxy",
          "--kubeconfig=/var/lib/kubelet/kube-proxy-config",
          "--hostname-override=9.111.255.21",
          "--feature-gates=SupportIPVSProxyMode=true",
          "--proxy-mode=ipvs",
          "--ipvs-min-sync-period=5s",
          "--ipvs-sync-period=5s",
          "--ipvs-scheduler=rr",
          "--cluster-cidr=10.1.0.0/16"
        ],
```

This is all of the services in my cluster.
```
[root@gyliu-icp-1 pods]# kc get svc --all-namespaces
NAMESPACE     NAME         TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)                               AGE
default       kubernetes   ClusterIP   10.0.0.1     <none>        443/TCP                               1h
default       my-nginx     NodePort    10.0.0.141   <none>        80:31043/TCP                          37m
kube-system   cloudantdb   ClusterIP   10.0.0.126   <none>        4369/TCP,5984/TCP,6984/TCP,9000/TCP   1h
kube-system   ds-hs        ClusterIP   None         <none>        4369/TCP,5984/TCP,6984/TCP,9000/TCP   1h
kube-system   kube-dns     ClusterIP   10.0.0.10    <none>        53/UDP,53/TCP                         1h
```

But the ipvs table was not updated
```
[root@gyliu-icp-1 pods]# ipvsadm -ln
IP Virtual Server version 1.2.1 (size=4096)
Prot LocalAddress:Port Scheduler Flags
  -> RemoteAddress:Port           Forward Weight ActiveConn InActConn
TCP  10.0.0.1:443 rr persistent 10800
  -> 9.111.255.21:8001            Masq    1      4          3
TCP  10.0.0.10:53 rr
  -> 10.1.198.129:53              Masq    1      0          0
TCP  10.0.0.126:4369 rr
TCP  10.0.0.126:5984 rr
TCP  10.0.0.126:6984 rr
TCP  10.0.0.126:9000 rr
UDP  10.0.0.10:53 rr
  -> 10.1.198.129:53              Masq    1      0          188
```

From above, we can see that the ip such as `TCP  10.0.0.126:9000 rr` do not have destination pod,  also some services, such as `my-nginx` was not added to the ipvs table.

/cc @m1093782566 want to get some comments from you: What is the quality of ipvs in 1.8.2 with calico? Can I do some performance test based on this? Seems not..

/sig network
/area ipvs

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by m1093782566** [comment]
@gyliu513 Thanks for reporting it.

> From above, we can see that the ip such as TCP 10.0.0.126:9000 rr do not have destination pod, 

What's the output of 

```
kubectl get endpoints --all-namespaces
```

> also some services, such as my-nginx was not added to the ipvs table.

I feel like you have reported a NodePort bug yesterday? The eth{i} issue?

**Comment by m1093782566** [comment]
> What is the quality of ipvs in 1.8.2 with calico? 

Not sure. Since IPVS-based kube-proxy is still in alpha. I always suggest people check the HEAD.

**Comment by gyliu513** [comment]
@m1093782566 it is not eth{i} issue as I have updated all of the nic to eth{i}.

```
[root@gyliu-icp-1 ~]# kc get endpoints --all-namespaces
NAMESPACE     NAME                      ENDPOINTS                                                           AGE
default       kubernetes                9.111.255.21:8001                                                   8h
default       my-nginx                  10.1.55.194:80                                                      6h
kube-system   cloudantdb                                                                                    8h
kube-system   ds-hs                     10.1.198.131:5984,10.1.198.131:4369,10.1.198.131:6984 + 1 more...   8h
kube-system   kube-controller-manager   <none>                                                              8h
kube-system   kube-dns                  10.1.198.129:53,10.1.198.129:53                                     8h
kube-system   kube-scheduler            <none>                                                              8h
```

How can the `endpoints` help me to debug this issue?

**Comment by m1093782566** [comment]
v1.8.2 do has such bug, I believe it's fixed in v1.9.

I am going to close this issue. Please re-open if it still persisit...
